### ISSUE TITLE
Progress does not work on Windows using "Git Bash"

### ISSUE THREAD

**Issue by masterholdy** [rclone/rclone#3531]
#### What is the problem you are having with rclone?
Im using rclone via Git Bash to Sync Data from my PC to my Gdrive, if im using the -P Parameter im getting the following error:

> 2019/09/12 18:56:15 ERROR : Failed to start progress: windows terminal: initiali                                                                                                                                                                                            zation failed

if im using cmd instead of git bash everything works like it should 
but i pref git bash because i can simply use while loops etc.

#### What is your rclone version (output from `rclone version`)

> rclone v1.49.2
> - os/arch: windows/amd64
> - go version: go1.12.3
> 

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

Windows 10, 64 bit

####  Which cloud storage system are you using? (eg Google Drive)

Google Drive

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
> ls -d * | while read x; do rclone sync "$x" "gdrive:/xyz/$x" -P; done

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
`$ rclone sync -vv "zzz" "gdrive:/xyz/zzz" -P`

> 2019/09/12 19:09:50 DEBUG : rclone: Version "v1.49.2" starting with parameters ["C:\\rclone\\rclone.exe" "sync" "-vv" "zzz" "gdrive:/xyz/zzz" "-P"]
2019/09/12 19:09:50 DEBUG : Using config file from "C:\\Users\\xxx\\.config\\rclone\\rclone.conf"
2019/09/12 19:09:51 ERROR : Failed to start progress: windows terminal: initialization failed

multiple file checks 
> 2019/09/12 19:09:51 DEBUG : xxx: Size and modification time the same (differ by -487.9µs, within tolerance 1ms)
2019/09/12 19:09:51 DEBUG : xxx: Unchanged skipping

`2019/09/12 19:09:52 INFO  : Google drive root 'xyz/zzz': Waiting for checks to finish`

> 2019/09/12 19:09:52 DEBUG : 11 go routines active
2019/09/12 19:09:52 DEBUG : rclone: Version "v1.49.2" finishing with parameters ["C:\\rclone\\rclone.exe" "sync" "-vv" "zzz" "gdrive:/xyz/zzz" "-P"]

**Comment by ncw** [comment]
So the sync works just without progress...

Is git bash a faithful vt100 terminal?

I guess the Unix code will work for it so it is just a matter of detecting that we are running in git bash.

 Any ideas?

**Comment by masterholdy** [comment]
So the sync works just without progress...

> exactly

Is git bash a faithful vt100 terminal?

> i dont know, sorry

I guess the Unix code will work for it so it is just a matter of detecting that we are running in git bash.

Any ideas?

- maybe using a flag like --bash

- or maybe run an unix cmd and check the result 

```
        boolean bash;
        try {
            Process p = Runtime.getRuntime().exec("ls .");
            bash = true;
        } catch (IOException e) {
            bash = false;
        }

        System.out.println( "Bash: \t" + (bash? "true" : "false"));
```
cmd: 

> C:\workspace\intelli\test\target>java -jar test-1.0-SNAPSHOT.jar
> Bash:   false

git bash

> $ java -jar test-1.0-SNAPSHOT.jar
> Bash:   true

worked in java 
both results are not pretty obv.
i personaly would pref flag over the second solution

**Comment by ncw** [comment]
I think looking at the `TERM` environment variable would be the right thing to do...

Git bash sets `TERM=xterm` which is a pretty clear signal that rclone is running in a functional terminal...

Try this - it works for me in cmd, powershell and git bash

https://beta.rclone.org/branch/v1.49.0-043-ge09a4ff0-fix-3531-git-bash-beta/ (uploaded in 15-30 mins)

**Comment by masterholdy** [comment]
is works now ty! :)

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.50
